### ISSUE TITLE
Singleton member with type info not set in release mode

### ISSUE THREAD

**Issue by borlox** [godotengine/godot#27099]
**Godot version:** 3.1

**OS/device including version:** 

I observed the behaviour in projects exported for HTML5 and Windows 10.

**Issue description:**

My (simplified) project has an AutoLoad singleton with a variable:
```
# global.gd
var value: float = 0.0
```

In my main scene I set the value in `_ready()`:
```
# Node2D.gd

func _ready():
	global.value = 42 # note: using an int value here, not 42.0!
	print(global.value)
	
	var local: float = 42
	print(local)
```

When running from the editor, the output is `42` for both prints, as expected. When I export the project (e.g. as HTML) with debugging disabled, the output (in the web console) is:
```
0
42
```

It seems that in release mode setting the variable to an integer value does not lead to a type-cast, but instead the action is ignored and the variable not changed.

**Steps to reproduce:**

- Add an auto load singleton script with a variable with a `float` type hint.
- Set this variable to an integer value in some scene's script.
- No type casting is performed and the variable is not set.

**Minimal reproduction project:**

[BugReport.zip](https://github.com/godotengine/godot/files/2971915/BugReport.zip)

**Comment by bojidar-bg** [comment]
CC @vnen, in case you have ideas what might be causing this.

**Comment by vnen** [comment]
It's the same problem as #24035

**Comment by vnen** [comment]
This was solved by #34918.
